src/ert/_c_wrappers/enkf/enkf_main.py

```diff
@@ -410,63 +404,62 @@ def sample_prior(
         state_map = storage.getStateMap()
-        for realization_nr in active_realizations:
-            current_status = state_map[realization_nr]
-            for parameter in parameters:
-                config_node = self.ensembleConfig().getNode(parameter)
-                enkf_node = EnkfNode(config_node)
-
-                if config_node.getUseForwardInit():
-                    continue
-                if (
-                    not enkf_node.has_data(storage, NodeId(0, realization_nr))
-                    or current_status == STATE_LOAD_FAILURE
-                ):
-                    rng = self.realizations[realization_nr]
-                    impl_type = enkf_node.getImplType()
-
-                    if impl_type == ErtImplType.GEN_KW:
-                        gen_kw_node = enkf_node.asGenKw()
-                        if len(gen_kw_node) > 0:
-                            if config_node.get_init_file_fmt():
-                                df = pd.read_csv(
-                                    config_node.get_init_file_fmt() % realization_nr,
-                                    delim_whitespace=True,
-                                    header=None,
-                                )
-                                # This means we have a key: value mapping in the
-                                # file otherwise it is just a list of values
-                                if df.shape[1] == 2:
-                                    # We need to sort the user input keys by the
-                                    # internal order of sub-parameters:
-                                    keys = [val[0] for val in gen_kw_node.items()]
-                                    df = df.set_index(df.columns[0])
-                                    vals = df.reindex(keys).values.flatten()
-                                else:
-                                    vals = df.values.flatten()
-                            else:
-                                vals = rng.standard_normal(len(gen_kw_node))
-                            s = io.BytesIO()
-                            # The first element is time_t (64 bit integer), but
-                            # it is not used so we write 0 instead - for
-                            # padding purposes.
-                            s.write(struct.pack("Qi", 0, int(ErtImplType.GEN_KW)))
-                            s.write(vals.tobytes())
-
-                            _clib.enkf_fs.write_parameter(
-                                storage,
-                                config_node.getKey(),
-                                realization_nr,
-                                s.getvalue(),
-                            )
-
-                    else:
-                        _clib.enkf_state.state_initialize(
-                            enkf_node,
-                            storage,
-                            realization_nr,
-                        )
 
+        for parameter in parameters:
```

Comment:
This loop is very long but with clearly defined sections, could we split out some of the things in this loop to separate functions for readability?

Reply:
Yes, I contemplated this, but did not know where to put it. Did not want to add more stand-along functions inside `enkf_main.py` as it is starting to become a lot. I have added a suggested approach, but happy to hear other suggestions.

Reply:
Seems very reasonable, just add the return type as well and I think this is good to go!